### ISSUE TITLE
Allow changing dataset in Pipelines

### ISSUE THREAD

**Issue by AlekseySh** [OML-Team/open-metric-learning#528]
Let's make dataset part of Pipeline flexible, so we can easily work with modalities other than images.
Waiting for: https://github.com/OML-Team/open-metric-learning/issues/522

**Comment by leoromanovich** [comment]
I've started work. WIP PR will be attached shortly.

**Comment by leoromanovich** [comment]
Start work here: #585

**Comment by AlekseySh** [comment]
Let's start with the first PR, where we don't add texts support, but refactor the way of processing images datasets

**Registry**
```python
REGISTRY = {"IQGLD": ..., "ILD": ...}

def get_dataset(cfg) -> IQGLD | ILD:
...
```

+ update `oml.confifgs` and `test_registry` + update doc about customisation of pipeline: [link](https://github.com/OML-Team/open-metric-learning/tree/main/pipelines/features_extraction#customization)

**Config.yaml**
```python
dataset_train:
  name: image_label_dataset
  args:
    df: path_to_df/train.csv
    cache_size: 1999
    transform:
      name: hypvit_torch
      args:
        image_size: 224

dataset_val:
  ...
```

**Back compatibility**
```python
def convert_to_new_format_if_needed(cfg):
     if "dataset_root" and "transforms_train" and ... in cfg:
        cfg["dataset_train"] = {"name": "image_label_dataset", args: {"df": ..., "transform": }}
    # don't forget to delete refactored keys
     ....

def parse_datasets(cfg):
     dataset_train = get_dataset_by_cfg(cfg["dataset_train"])
     assert train_dataset is ILD

     dataset_val = get_dataset_by_cfg(cfg["dataset_val"])
     assert dataset_val is IQGLD

     if isinstance(dataset_train, ImageLD) and isinstance(dataset_val, ImageQGLD):
         df_train, df_val = dataset_train.df, dataset_val.df
         df_train, df_val = enumerate(df_train, df_val)
         dataset_train.df = df_train
         dataset_val.df = df_val

      # check labels are enums without gaps (arange(T+V))
     labels_train = dataset_train.get_labels()
     labels_val = dataset_val.get_labels()
     raise check_consistency(labels_train, labels_val)  # write a test that fails here

     return dataset_train, dataset_val



def extractor_training_pipeline():
        cfg = dictconfig_to_dict(cfg)
        cfg = convert_to_new_format_if_needed(cfg)

        dataset_train, dataset_val = parse_datasets(cfg)

```

**UpdateDatasets**
```
Dataset(df: pd.Dataframe) -> Dataset(df: Union[pd.Dataframe, Path])
```

**Update mock dataset and pipelines test**
```python
@hydra.main(config_path="configs", config_name="train_postprocessor.yaml", version_base=HYDRA_BEHAVIOUR)
def main_hydra(cfg: DictConfig) -> None:
    cfg = dictconfig_to_dict(cfg)
    df_tain_path, df_val_path = download_mock_dataset(MOCK_DATASET_PATH, return_paths=True) 
    cfg["dataset_train"]["args"]["df"] = df_tain_path
    cfg["dataset_val"]["args"]["df"] = df_val_path
    extractor_training_pipeline(cfg)


if __name__ == "__main__":
    main_hydra()
```
